src/slashcommands/ISlashCommand.ts

```diff
@@ -0,0 +1,6 @@
+// just guessing
+export interface ISlashCommand {
+    command: string;
+    pattern: string;
+    function: () => boolean;
```

Comment:
I believe that this comment doesn't quite refer to the `ISlashCommand.ts`, but to the `execute` method, right? Well, it's a command-pattern and commands have the benefit of being nicely enhanceable via decorators. However, this leads to not very verbose signatures. and `IExecutionContext` would have to be very generically typed.
With respect to the result: That's a very fundamental question: "How should mutations be performed". 

- Up to now, I thought we wanted to inject objects which allow the Rocketlet to **perform** this mutations. All methods would actually return `void` . In my experience, it is still nice to return a "failed and reasons why it failed".
- If you wanted to return e. g. messages to be sent (and not have an accessor which enables the Rocketlet to send the message), you can export a command which the framework shall execute afterwards. But again, this would be quite generic and whether the command can actually be executed depends on the implementation of the "return command executor". If this one rejects (one or more) of the modifying commands, the Rocketlet had no option to react on that.

Thus, I`d go for the first approach.

Was that understandable?